internal/registry/registry.go

```diff
@@ -113,8 +112,12 @@ func (r *registry) LastLayer(ctx context.Context, image string, po *kmmv1beta1.P
 func (r *registry) VerifyModuleExists(layer v1.Layer, pathPrefix, kernelVersion, moduleFileName string) bool {
 	// in layers headers, there is no root prefix
 	fullPath := filepath.Join(strings.TrimPrefix(pathPrefix, "/"), modulesLocationPath, kernelVersion, moduleFileName)
-	_, err := r.GetHeaderDataFromLayer(layer, fullPath)
-	return err == nil
+	_, readerCloser, err := r.getHeaderReaderFromLayer(layer, fullPath)
+	if err != nil {
+		return false
+	}
+	readerCloser.Close()
+	return true
```

Comment:
```suggestion
	_, readerCloser, err := r.getHeaderReaderFromLayer(layer, fullPath)
	defer readerCloser.Close()
        return err == nil
```

Reply:
I cannot use your suggestion since  in case getHeaderReaderFromLayer returns an error, readerCloser will be nil.